mutation_compactor.hh

```diff
@@ -408,7 +408,8 @@ public:
 
         if (!cr.empty()) {
             partition_is_not_empty(consumer);
-            _stop = consumer.consume(std::move(cr), t, is_live);
+            auto stop = consumer.consume(std::move(cr), t, is_live);
+            _stop = stop && !sstable_compaction() ? stop_iteration::yes : stop_iteration::no;
```

Comment:
this could be simplified as `stop_iteration(stop && !sstable_compaction())`

Reply:
I don't understand why we need these games here? Can't the consumer just not return `stop_iteration::yes` if it doesn't really want to stop the consumption?

Reply:
Followed convention. mutation_compactor::consume_end_of_partition() ignores stop request if mutation_compactor::sstable_compaction() returns true. So I think the patch is valid, as a defense mechanism, but your suggestion can also happen, meaning they're not mutually exclusive.